sdk/notificationhubs/notification-hubs/src/notificationHubsClient.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationOptions, ServiceClient } from "@azure/core-client";
+import { createHttpHeaders, createPipelineRequest } from "@azure/core-rest-pipeline";
+import { createTokenProviderFromConnectionString, parseEndpoint, parseNotificationHubsConnectionString } from "./utils/connectionStringUtils";
+import { NotificationHubsClientOptions } from "./models";
+import { SasTokenProvider } from "@azure/core-amqp";
+import { tracingClient } from "./utils/tracing";
+
+const API_VERSION = "2020-06";
+
+export interface NotificationHubResponse {
+  trackingId?: string;
+  correlationId?: string;
+  location?: string;
+}
+
+export class NotificationHubsClient extends ServiceClient {
+  private _baseUrl: { host: string; hostname: string; port?: string } ;
+  private _hubName: string;
+  private _sasTokenProvider: SasTokenProvider;
+
+  constructor(connectionString: string, hubName: string, options: NotificationHubsClientOptions = {}) {
+    super(options);
+    this._hubName = hubName;
+
+    const parsedConnection = parseNotificationHubsConnectionString(connectionString);
+    this._baseUrl = parseEndpoint(parsedConnection.endpoint);
+    this._sasTokenProvider = createTokenProviderFromConnectionString(connectionString);
+  }
+
+  public deleteInstallation(installationId: string, options: OperationOptions = {}): Promise<NotificationHubResponse> {
+    return tracingClient.withSpan("NotificationHubsClient-deleteInstallation", options, async (updatedOptions) => {
+      const endpoint = `${this._baseUrl.hostname.replace("sb://", "https://")}/${this._hubName}/installations/${installationId}?api-version=${API_VERSION}`;
+      const authorization = this._sasTokenProvider.getToken(this._baseUrl.hostname);
+      const headers = createHttpHeaders();
+      headers.set("Authorization", authorization.token);
+      headers.set("x-ms-version", API_VERSION);
+
+      const request = createPipelineRequest({
+        ...updatedOptions.tracingOptions,
+        ...updatedOptions.requestOptions,
+        url: endpoint,
+        abortSignal: updatedOptions.abortSignal,
+        method: "DELETE",
+        headers
+      });
+
+      const response = await this.sendRequest(request);
+      if (response.status !== 204) {
+        // TODO: throw special error
+        throw new Error(`deleteInstallation failed with ${response.status}`);
```

Comment:
custom error type with type guard? :)

Reply:
How do you return back with those that can be retried and those that cannot?

Reply:
Hmm... well when it's something we auto-retry (using the standard retry policies) that doesn't really get exposed to the consumer. If it's something *they* need to retry.... probably a custom error with useful properties that implements an interface that is compatible with a retry operation method?

Reply:
For example, if the customer gets a 403, 429, or a 503, I would expect them to retry the operation, I just need a way to expose that back so it retries for them using a policy.

Reply:
429 we should retry for automatically, the other two it depends on how you model it. Our default behavior for unknown codes (e.g. not specified in swagger) is always to throw a RestError and let the user handle it. Sometimes services model things differently or have custom error shapes tied to specific codes.

Reply:
Ok, will investigate for the next release.